operator/pkg/gateway-api/cell.go

```diff
@@ -44,6 +45,7 @@ var requiredGVK = []schema.GroupVersionKind{
 	gatewayv1beta1.SchemeGroupVersion.WithKind("referencegrants"),
 	gatewayv1alpha2.SchemeGroupVersion.WithKind("grpcroutes"),
 	gatewayv1alpha2.SchemeGroupVersion.WithKind("tlsroutes"),
+	mcsapiv1alpha1.SchemeGroupVersion.WithKind("serviceimports"),
```

Comment:
Adding the CRD `ServiceImport` of the MCS API to the list of required kinds introduces a "hard dependency" from Cilium's Gateway API implemention to the MCS API. Hence if MCS API's `ServiceImport` isn't present in the cluster, the Gateway API controller won't be enabled at all - even if the Gateway API CRDs are installed.

This is indicated by the following message at the startup of the Cilium Operator

```
level=error msg="Required GatewayAPI resources are not found, please refer to docs for installation instructions" error="customresourcedefinitions.apiextensions.k8s.io \"serviceimports.multicluster.x-k8s.io\" not found" subsys=gateway-api
```

I think we should support the use case where someone only wants to deploy the Gateway API CRDs without having the MCS API CRDs installed.

But by simply removing the resource from this list, the reconciliation will fail when the underlying informers try to watch the `ServiceImport`.

e.g. 

```
level=error msg="failed to get informer from cache" error="failed to get API group resources: unable to retrieve the complete list of server APIs: multicluster.x-k8s.io/v1alpha1: the server could not find the requested resource" logger=controller-runtime.source.EventHandler subsys=controller-runtime
```

IMO - at best the presence of the CRD `ServiceImport` within the cluster would only enable the additional logic.

Reply:
> I think we should support the use case where someone only wants to install the Gateway API CRDs without having to install the MCS API CRDs.

Oh yes indeed thanks for the detailed explanations I assumed this was doing this already, will try to figure out we could have something optional :+1:

Reply:
Indeed it's actually very clear form the code since there the variable have `required` in its name, I must have missed that during a rebase sorry :sweat_smile:.

So I added a second commit doing just as you suggested, it check only once lazily for consistency between runtime checks and init checks (i.e.: decide if we want to watch the resource). However this leads me to add a `ResetServiceImportCRD` to reset this check during testing which I am not fully convinced, feel free if you have any suggestion :see_no_evil:

Reply:
Great catch @mhofstetter, I missed this. I agree we shouldn't hard block on MCS resources.

Reply:
The second commit LGTM overall, although I'm not sure if `routechecks` is the right package for this. Could you talk more about why you put it there?

Reply:
Hmm yes so I added it here because I also added a route check `CheckHasServiceImportSupport` so I thought it was the most convenient to do so but I can move it if you think that's appropriate.

Reply:
Agree that we should move this to helpers as this isn't an actual route check - even though it gets used by one.

see https://github.com/cilium/cilium/pull/28769#discussion_r1425082982 & https://github.com/cilium/cilium/pull/28769#discussion_r1425093056